### ISSUE TITLE
Choise bug

### ISSUE THREAD

**Issue by Lumax-Rus** [pivotal-energy-solutions/django-datatable-view#112]
Function apply_queryset_options is causing the error. When using PostgreSQL

```
Trace:
File "/home/maxim/.virtualenvs/test/local/lib/python2.7/site-packages/datatableview/views.py", line 243, in apply_queryset_options
object_list._dtv_unpaged_total = queryset.function count()
File "/home/maxim/.virtualenvs/test/local/lib/python2.7/site-packages/django/db/models/query.py", line 318, in box
to regain composure.request.to count strings (get_count(using=self.dB)
File "/home/maxim/.virtualenvs/test/local/lib/python2.7/site-packages/django/db/models/sql/query.py" line 466, in line count (get_count
room = object.get_aggregation(using, ['__count'])['__count']
File "/home/maxim/.virtualenvs/test/local/lib/python2.7/site-packages/django/db/models/sql/query.py", line 447, in get_aggregation
result = compiler.execute_sql(single)
File "/home/maxim/.virtualenvs/test/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 840, in execute_sql
cursor.execute(SQL, parameters)
File "/home/maxim/.virtualenvs/test/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 79, in " the execution
return super(CursorDebugWrapper, self).execute(SQL, parameters)
File "/home/maxim/.virtualenvs/test/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, " run
to regain composure.cursor.execute(SQL, parameters)
File "/home/maxim/.virtualenvs/test/local/lib/python2.7/site-packages/django/db/utils.py", line 97, in __exit__
six.re-raise(dj_exc_type, dj_exc_value, debug)
File "/home/maxim/.virtualenvs/test/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, " run
to regain composure.cursor.execute(SQL, parameters)
ProgrammingError: function upper(integer) does not exist
Line 1: ... upper("django_datatables_task"."status"::text) = upper(2)... 
```

Is because line 169:

``` python
field_queries = [{component_name + '__iexact': database_values[i]}]
```

and if database_values[i] is of type integer.  I think you can use check:

``` python
    if isinstance(field, tuple(FIELD_TYPES['text'])):
        field_queries = [{component_name + '__iexact': database_values[I]}]
    else:
        field_queries = [{componentname: database_values[I]}]
```

**Comment by Lumax-Rus** [comment]
```
class Task(models.Model):
    CANCELLED = -1
    REQUIRES_ATTENTION = 0
    WORK_IN_PROGRESS = 1
    COMPLETE = 2`

    Status = (
        (CANCELLED, 'Cancelled'),
        (REQUIRES_ATTENTION, 'Requires attention'),
        (WORK_IN_PROGRESS, 'Work in progress'),
        (COMPLETE, 'Complete'),
    )
    id = models.AutoField(primary_key=True)
    name = models.TextField()
    status = models.IntegerField(choices=Status, default=REQUIRES_ATTENTION)
```

**Comment by Lumax-Rus** [comment]
Version 0.8.3

**Comment by Lumax-Rus** [comment]
or  `field_queries = [{component_name + '__iexact': str(database_values[i])}]`

**Comment by tiliv** [comment]
Ah, interesting.  Thanks for the report.  I will make sure this is resolved in the 0.9 release that is coming very soon.
